gr-qtgui/python/qtgui/range.py.cmakein

```diff
@@ -234,6 +291,69 @@ class RangeWidget(QtWidgets.QWidget):
             self.notifyChanged(self.rangeType(value))
 
 
+    class EngSlider(QtWidgets.QWidget):
+        """ Creates the range using a counter and slider """
+        def __init__(self, parent, ranges, slot, rangeType=float):
+            QtWidgets.QWidget.__init__(self, parent)
+
+            self.first = True
+            self.rangeType = rangeType
+
+            # Slot to call in the parent
+            self.notifyChanged = slot
+
+            self.slider = RangeWidget.Slider(parent, ranges, self.sliderChanged, rangeType)
+            self.counter = RangeWidget.Eng(parent, ranges, self.counterChanged, rangeType)
+
+            # Need another horizontal layout to wrap the other widgets.
+            layout = Qt.QHBoxLayout()
+            layout.setContentsMargins(0, 0, 0, 0)
+            layout.addWidget(self.slider)
+            layout.addWidget(self.counter)
+            self.setLayout(layout)
+
+            # Flags to ignore the slider event caused by a change to the counter (and vice versa).
+            self.ignoreSlider = False
+            self.ignoreCounter = False
+            self.range = ranges
+
+        def sliderChanged(self, value):
+            """ Handles changing the counter when the slider is updated """
+            # If the counter was changed, ignore any of these events
+            if not self.ignoreSlider:
+                # convert Value to eng string
+                self.ignoreCounter = True
+                self.counter.setText(eng_notation.num_to_str(self.rangeType(value)))
+                self.notifyChanged(self.rangeType(value))
+            self.ignoreSlider = False
+
+        def counterChanged(self, value):
+            """ Handles changing the slider when the counter is updated """
+            # Get the current slider value and check to see if the new value changes it
+            current = self.slider.value()
+            print("counterChanged",value,"ign",self.ignoreCounter)
+            new = int(round(self.range.demap_range(value), 0))
+
+            # If it needs to change, ignore the slider event
+            # Otherwise, the slider will cause the counter to round to the nearest tick
+            if current != new:
+                    self.ignoreSlider = True
+                    self.slider.setValue(new)
+
+            if not self.ignoreCounter:
+                print("to notify",self.rangeType(value))
+                self.notifyChanged(self.rangeType(value))
+            self.ignoreCounter = False
+
+        def setValue(self, value):
+            """ Wrapper to handle changing the value externally """
+            self.counter.setText(eng_notation.num_to_str(value))
+            if self.first or True:
+                new = int(round(self.range.demap_range(value), 0))
+                self.ignoreSlider = True
+                self.slider.setValue(new)
+                self.first = False
+
 if __name__ == "__main__":
     from PyQt4 import Qt
```

Comment:
uh, err,  shouldn't this be Qt5?

Reply:
Probably ... but it's also not part of this PR :)

Reply:
Yes: not in this PR . There are some 16 files in current master that have "PyQt4" in them ... that's -really- strange & hopefully someone can get this fixed!